azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -278,7 +278,7 @@ private void updatePodStatus(AzPodStatusMetadata event) {
       // Emit EXECUTION_STOPPED flow event
       this.fireEventListeners(Event.create(executableFlow,
           EventType.FLOW_FINISHED, new EventData(executableFlow)));
-      ExecutionControllerUtils.restartFlow(executableFlow, originalStatus);
+      ExecutionControllerUtils.restartFlow(executableFlow, Status.EXECUTION_STOPPED);
```

Comment:
The context of why `execution_stopped` status is passed is not easy to understand here. If the intention was to make sure we get to the flow param part in the restartFlow method, can we just have a separate method in `ExecutionControllerUtils` and call that method?

Reply:
Instead of passing a Status.EXECUTION_STOPPED here, can we have a high-priority if in ExecutionControllerUtils.restartFlow checking:
1. currentStatus == EXECUTION_STOPPED && allow.restart.on.execution.stopped:true (should be placed after "options.isExecutionRetried()" check because we don't want to restart a flow twice)

So the flow can be still restarted there.